src/Microsoft.XmlSerializer.Generator/src/Microsoft/XmlSerializer/Generator/XmlSerializationWriter.cs

```diff
@@ -2080,6 +2080,14 @@ private void WriteValue(object value)
                     {
                         Writer.Write(((int)value).ToString(null, NumberFormatInfo.InvariantInfo));
                     }
+                    else if (type == typeof(TimeSpan))
```

Comment:
Yes, I compared the code in desktop and found the code in core is a little different from desktop, which all use if/else instead of switch/case. I don't know why they change the format when port the code in core. It is out of scope of this checkin. We can review later. Now I just need make sure it will generate the right code when type is TimeSpan.